### PR TITLE
Rename num_classes to classes across entire repo

### PR DESCRIPTION
**Pull request by ianjjohnson**
For consistency with Keras.applications, and common Keras convention.

@LukeWood @fchollet

### CODE REVIEW

**Comment by LukeWood** [comment]
Can we also do this for non_max_suppression and any other layers with a `num_classes` parameter while we're here?  Also lets create a keras-io pull request

**Comment by LukeWood** [comment]
Thanks @ianjjohnson for the swift PR.

https://github.com/keras-team/keras-io/blob/master/templates/keras_cv/index.md will also need to be updated, and perhaps several developer guides.

File: benchmarks/metrics/coco/mean_average_precision_bucket_performance.py [hunk 1]
```diff
@@ -23,9 +23,9 @@ def produce_random_data(include_confidence=False, num_images=128, num_classes=20
     images = []
     for _ in range(num_images):
         num_boxes = math.floor(25 * random.uniform(0, 1))
-        classes = np.floor(np.random.rand(num_boxes, 1) * num_classes)
+        classes_in_image = np.floor(np.random.rand(num_boxes, 1) * classes)
```
Comment (ianjjohnson) [inline benchmarks/metrics/coco/mean_average_precision_bucket_performance.py:26]: This is the only place where a meaningful rename of an existing variable had to occur. LMK what you think.

Comment (ianjjohnson) [reply benchmarks/metrics/coco/mean_average_precision_bucket_performance.py:26]: (Also in mean_average_precision_performance.py)

Comment (ianjjohnson) [reply benchmarks/metrics/coco/mean_average_precision_bucket_performance.py:26]: (Also in recall_performance.py)

File: keras_cv/layers/object_detection/non_max_suppression.py [hunk 2]
```diff
@@ -112,11 +112,11 @@ def call(self, predictions, images=None):
 
         # preparing the predictions for TF NMS op
         boxes = tf.expand_dims(predictions[..., :4], axis=2)
-        classes = tf.cast(predictions[..., 4], tf.int32)
+        targets = tf.cast(predictions[..., 4], tf.int32)
```
Comment (ianjjohnson) [inline keras_cv/layers/object_detection/non_max_suppression.py:115]: This also got renamed.

File: keras_cv/layers/object_detection/non_max_suppression.py [hunk 3]
```diff
@@ -112,11 +112,11 @@ def call(self, predictions, images=None):
 
         # preparing the predictions for TF NMS op
         boxes = tf.expand_dims(predictions[..., :4], axis=2)
-        classes = tf.cast(predictions[..., 4], tf.int32)
+        class_predictions = tf.cast(predictions[..., 4], tf.int32)
```
Comment (ianjjohnson) [inline keras_cv/layers/object_detection/non_max_suppression.py:115]: This also got renamed

**Review by LukeWood — APPROVED** [review]

**Pull request merged**